toml/decoder.py

```diff
@@ -284,6 +305,8 @@ def loads(s, _dict=dict, decoder=None):
                     j += 1
             except IndexError:
                 break
+            # store the comment info
+            decoder.store_comment_info(i, comment_info, comment_lines, original)
```

Comment:
I think this needs to go _before_ the while loop with sl[j] = ' ', or else you lose the comment text...